src/v/datalake/translation/translation_probe.cc

```diff
@@ -32,9 +32,44 @@ translation_probe::translation_probe(model::ntp ntp)
         register_created_files_metrics();
         register_invalid_record_metric();
         register_throughput_metrics();
+        register_lag_metrics();
     }
 }
 
+void translation_probe::register_lag_metrics() {
+    namespace sm = ss::metrics;
+    std::vector<sm::label_instance> labels{
+      namespace_label(_ntp.ns()),
+      topic_label(_ntp.tp.topic()),
+      partition_label(_ntp.tp.partition()),
+    };
+    _public_metrics->add_group(
```

Comment:
The general rule is that internal metrics are a superset of public metrics (sometimes with different names, etc) - is this metric only on public metrics? If so I think it should go in both places.

Reply:
Right now all iceberg metrics are public only. I'll add a followup PR to also mirror to internal metrics if there is no other feedback